drivers/adc/adc_ads131m02.c

```diff
@@ -0,0 +1,738 @@
+	sys_put_be16(write_data, &tx_buf[3]);
+
+	return ads131m02_transceive(dev, tx_buf, sizeof(tx_buf),
+				    rx_buf, sizeof(rx_buf));
+
+}
+
+static inline int ads131m02_configure_gain(const struct device *dev,
+					   const struct adc_channel_cfg *channel_cfg)
+{
+	uint16_t gain_cfg = 0;
+
+	switch (channel_cfg->gain) {
+	case ADC_GAIN_1:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_1);
+		break;
+	case ADC_GAIN_2:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_2);
+		break;
+	case ADC_GAIN_4:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_4);
+		break;
+	case ADC_GAIN_8:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_8);
+		break;
+	case ADC_GAIN_16:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_16);
+		break;
+	case ADC_GAIN_32:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_32);
+		break;
+	case ADC_GAIN_64:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_64);
+		break;
+	case ADC_GAIN_128:
+		gain_cfg = ADS131M02_GET_GAIN(channel_cfg->channel_id,
+					      ADS131M02_GAIN_128);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ads131m02_reg_write(dev, ADS131M02_GAIN_REG, gain_cfg);
+}
+
+static inline int ads131m02_acquistion_time(uint16_t acq_time)
+{
+	uint16_t acq_value = ADC_ACQ_TIME_VALUE(acq_time);
+
+	if (acq_time == ADC_ACQ_TIME_DEFAULT) {
+		return ADS131M02_OSR_1024_MASK;
+	}
+
+	if (ADC_ACQ_TIME_UNIT(acq_time) != ADC_ACQ_TIME_TICKS) {
+		return -EINVAL;
+	}
+
+	if (acq_time == ADC_ACQ_TIME_MAX) {
+		return ADS131M02_OSR_16384_MASK;
+	}
+
+	switch (acq_value) {
+	case ADS131M02_DR_250:
+		return ADS131M02_OSR_16384_MASK;
+	case ADS131M02_DR_500:
+		return ADS131M02_OSR_8192_MASK;
+	case ADS131M02_DR_1k:
+		return ADS131M02_OSR_4096_MASK;
+	case ADS131M02_DR_2k:
+		return ADS131M02_OSR_2048_MASK;
+	case ADS131M02_DR_4k:
+		return ADS131M02_OSR_1024_MASK;
+	case ADS131M02_DR_8k:
+		return ADS131M02_OSR_512_MASK;
+	case ADS131M02_DR_16k:
+		return ADS131M02_OSR_256_MASK;
+	case ADS131M02_DR_32k:
+		return 0;
+	default:
+		return -EINVAL;
+	}
+
+	return -EINVAL;
+}
+
+static int ads131m02_setup(const struct device *dev,
+			   const struct adc_channel_cfg *channel_cfg)
+{
+	int ret = 0;
+	uint16_t enable = 0;
+	uint8_t read_data[3] = {0};
+
+	ret  = ads131m02_reg_read(dev, ADS131M02_CLOCK_REG, read_data,
+				  sizeof(read_data));
+	if (ret != 0) {
+		return ret;
+	}
+
+	enable = sys_get_be16(read_data);
+	switch (channel_cfg->channel_id) {
+	case ADC_CHANNEL_0:
+		enable |= ADS131M02_CHANNEL0_ENABLE;
+		break;
+	case ADC_CHANNEL_1:
+		enable |= ADS131M02_CHANNEL1_ENABLE;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	ret = ads131m02_acquistion_time(channel_cfg->acquisition_time);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (ret > 0) {
+		enable |= ret;
+	}
```

Comment:
uhm, this looks funny, what if it's ret == 0? could maybe use a comment though it may not be a bad idea to pass `&enable` to `ads131m02_acquistion_time` and let the function modify it instead and drop this one